src/cli/install.ps1

```diff
@@ -1,176 +1,220 @@
+if (!(Test-Path -LiteralPath $installationPath -PathType Container)) {
+    Write-Verbose "Installation directory does not exist."
+    try {
+        Write-Verbose "Create installation directory at `"$installationPath`""
+        New-Item $installationPath -Type Directory | Out-Null
+    } catch {
+        $errorMessage = $PSItem.Exception.Message
+        Write-Error -Message "Error with environment variable BUN_INSTALL. $errorMessage" -Exception $PSItem.Exception -ErrorAction Stop
+    }
+}
+
+# Add or update environment variable
+Write-Verbose "Add user environment variable BUN_HOME=$installationPath"
+[Environment]::SetEnvironmentVariable("BUN_HOME", $installationPath, "User")
+$env:BUN_HOME = $installationPath
+if ([Environment]::GetEnvironmentVariable("PATH", "User") -notlike "*%BUN_HOME%*") {
+    $oldUserPATH = [Environment]::GetEnvironmentVariable("PATH", "User")
+    $newUserPATH = "$oldUserPATH;%BUN_HOME%".Replace(";;", ";")
+    [Environment]::SetEnvironmentVariable("PATH", $newUserPATH, "User")
+}
+$env:Path = [Environment]::ExpandEnvironmentVariables(
+    [Environment]::GetEnvironmentVariable("Path", "Machine") + 
+    ";" + 
+    [Environment]::GetEnvironmentVariable("Path", "User")
+)
+#endregion
+
+#region Bun Download
+$installer = New-TemporaryFile
+
+if ($ForceBaseline) {
+    $file = "bun-windows-x64-baseline.zip"
+} else {
+    $file = "bun-windows-x64.zip"
+}
+
+if ($PSCmdlet.ParameterSetName -eq "Release") {
+    $url = "https://github.com/oven-sh/bun/releases/download/$($Release.ToLower())/$file"
+} elseif ($PSCmdlet.ParameterSetName -eq "Version") {
+    $url = "https://github.com/oven-sh/bun/releases/download/bun-v$Version/$file"
+}
+
+try {
+    Invoke-WebRequest $url -OutFile $installer
+} catch {
+    Remove-Item $installer
+    $errorMessage = $PSItem.Exception.Message
+    Write-Error -Message "Error downloading bun. $errorMessage" -Exception $PSItem.Exception -ErrorAction Stop
+}
+#endregion
+
+#region Bun Installation
+Expand-Archive -LiteralPath $installer -DestinationPath $installationPath\temp -Force
+Move-Item "$installationPath\temp\$($file.Split(".")[0])\bun.exe" $installationPath -Force
+Remove-Item -LiteralPath "$installationPath\temp\$($file.Split(".")[0])", "$installationPath\temp\"
+
+try {
+    New-Item -ItemType HardLink -Path "$installationPath\bunx.exe" -Target "$installationPath\bun.exe" -Force | Out-Null
+} catch {
+    Write-Warning "Could not create a hard link for bunx, falling back to a cmd script`n"
+    Set-Content -Path "$installationPath\bunx.cmd" -Value "@%~dp0bun.exe x %*"
+}
+#endregion
+
+#region Installation Verification
+$p = Start-Process $installationPath\bun.exe -ArgumentList "--revision" -NoNewWindow -Wait -PassThru -RedirectStandardOutput NUL
+switch ($p.ExitCode) {
+    0 { continue }
+    # STATUS_ILLEGAL_INSTRUCTION
+    1073741795 {
+        Write-Error ("Validation failed. Bun is not compatible with your CPU. This should have been detected before downloading.`n" +
```

Comment:
imo this should report the error but then re-run the installer with the proper download.